### PR TITLE
client-go: start fresh with owner file

### PR DESCRIPTION
**Pull request by sttts**
After client-go's OWNER file has grown to 50ish many reviewers, It's time to start fresh and re-add those who really want to review. Please don't be shy to come forward if you want to stay on that list. I will re-add you.

### CODE REVIEW

**Comment by sttts** [comment]
@kubernetes/sig-api-machinery-pr-reviews

File: staging/src/k8s.io/client-go/OWNERS [hunk 1]
```diff
@@ -7,39 +7,10 @@ approvers:
 - sttts
 - liggitt
 reviewers:
-- thockin
+- caesarxuchao
+- deads2k
+- krousey
```
[the remark this answers is not in the document]
Comment (sttts) [reply staging/src/k8s.io/client-go/OWNERS]: done

**Comment by sttts** [comment]
Addressed @lavalamp's comment. @deads2k list is fine with you?

File: staging/src/k8s.io/client-go/OWNERS [hunk 2]
```diff
@@ -1,45 +1,15 @@
...
-- mikedanese
+- deads2k
+- lavalamp
 - liggitt
-- nikhiljindal
-- gmarek
-- erictune
-- davidopp
-- pmorie
+- smarterclayton
```
Comment (deads2k) [inline staging/src/k8s.io/client-go/OWNERS]: Remove clayton from reviewers, but leave as approver. 

Add @soltysh as a reviewer (not approver).

**Comment by deads2k** [comment]
> Addressed @lavalamp's comment. @deads2k list is fine with you?

commented

File: staging/src/k8s.io/client-go/OWNERS (hunk 2 above)
Comment (sttts) [reply staging/src/k8s.io/client-go/OWNERS]: done

**Pull request merged**